libraries/exoplayer/src/main/java/androidx/media3/exoplayer/source/MediaSourceEventListener.java

```diff
@@ -47,12 +47,37 @@ public interface MediaSourceEventListener {
    *     LoadEventInfo#responseHeaders} will be empty.
    * @param mediaLoadData The {@link MediaLoadData} defining the data being loaded.
    */
+  @Deprecated
   default void onLoadStarted(
       int windowIndex,
       @Nullable MediaPeriodId mediaPeriodId,
       LoadEventInfo loadEventInfo,
       MediaLoadData mediaLoadData) {}
 
+  /**
+   * Called when a load begins.
+   *
+   * @param windowIndex The window index in the timeline of the media source this load belongs to.
+   * @param mediaPeriodId The {@link MediaPeriodId} this load belongs to. Null if the load does not
+   *     belong to a specific media period.
+   * @param loadEventInfo The {@link LoadEventInfo} corresponding to the event. The value of {@link
+   *     LoadEventInfo#uri} won't reflect potential redirection yet and {@link
+   *     LoadEventInfo#responseHeaders} will be empty.
+   * @param mediaLoadData The {@link MediaLoadData} defining the data being loaded.
+   * @param retryCount The number of failed attempts since {@link #onLoadStarted} was called (this
+   *     is zero for the first load attempt).
+   */
+  default void onLoadStarted(
+      int windowIndex,
+      @Nullable MediaPeriodId mediaPeriodId,
+      LoadEventInfo loadEventInfo,
+      MediaLoadData mediaLoadData,
+      int retryCount) {
+    if (retryCount == 0) {
+      onLoadStarted(windowIndex, mediaPeriodId, loadEventInfo, mediaLoadData);
```

Comment:
I think we should remove this attempted magic redirecting and keep it all in the `MediaSource` implementations.

This doesn't work when the call is redirected from `MediaSourceEventListener` to `AnalyticsListener` via `AnalyticsCollector` which overrides both the new and old `MediaSourceEventListener` methods (hence disabling this magic). `AnalyticsCollector` **must** override both methods for compat with custom `MediaSource` impls that aren't yet dispatching the new method.